avocado/utils/config_change_validation.py

```diff
@@ -0,0 +1,205 @@
+"""
+
+SUMMARY
+------------------------------------------------------
+Config change validation: Utility for testing config file changes.
+
+Copyright: 2012 Intra2net AG
+
+
+INTERFACE
+------------------------------------------------------
+
+"""
+
+import os
+import shutil
+
+# avocado utilities
+from avocado.utils import process
+
+
+def get_temp_file_path(file_path):
+    """
+    Generates a temporary filename.
+
+    :param str file_path: file path prefix
+    :returns: appended file path
+    :rtype: str
+    """
+    return file_path + '.tmp'
+
+
+def make_temp_file_copies(file_paths):
+    """
+    Creates temporary copies of the provided files.
+
+    :param file_paths: copied file paths
+    :type file_paths: [str]
+    """
+    for file_path in file_paths:
+        temp_file_path = get_temp_file_path(file_path)
+        shutil.copyfile(file_path, temp_file_path)
+
+
+def del_temp_file_copies(file_paths):
+    """
+    Deletes all the provided files.
+
+    :param file_paths: deleted file paths (their temporary versions)
+    :type file_paths: [str]
+    """
+    for file_path in file_paths:
+        temp_file_path = get_temp_file_path(file_path)
+        os.remove(temp_file_path)
+
+
+def parse_unified_diff_output(lines):
+    """
+    Parses the unified diff output of two files.
+
+    :param lines: diff lines
+    :type lines: [str]
+    :returns: pair of adds and removes, where each is a list of trimmed lines
+    :rtype: ([str], [str])
+    """
+    adds = []
+    removes = []
+    for line in lines:
+        # ignore filepaths in the output
+        if (len(line) > 2 and (line[:3] == "+++" or
+                               line[:3] == "---")):
+            continue
+        # ignore line range information in the output
+        elif len(line) > 1 and line[:2] == "@@":
+            continue
+        # gather adds
+        elif len(line) > 0 and line[0] == "+":
+            added_line = line[1:].lstrip().rstrip()
+            if len(added_line) == 0:
+                continue
+            adds = adds + [added_line]
+        # gather removes
+        elif len(line) > 0 and line[0] == "-":
+            removed_line = line[1:].lstrip().rstrip()
+            if len(removed_line) == 0:
+                continue
+            removes = removes + [removed_line]
+    return (adds, removes)
+
+
+def extract_config_changes(file_paths, compared_file_paths=None):
```

Comment:
The general intended use case is for config files as the name of the entire module implies as well. The suggested use cases are for when one wants to test changes on a system took effect but indeed, it could be generalized to any text files on the system. Do you suggest some better naming and do you think it could be used for much more than config and/or text files?